File: src/main/scala/io/izzel/taboolib/util/asm/AsmClassLoader.java
package io.izzel.taboolib.util.asm;

public class AsmClassLoader extends ClassLoader {

    private static final class AsmClassLoaderHolder {
        private static AsmClassLoader instance = new AsmClassLoader();
    }

    public static AsmClassLoader getInstance() {
        return new AsmClassLoader();
    }

    private AsmClassLoader() {
        super(AsmClassLoader.class.getClassLoader());
    }

    public static Class<?> createNewClass(String name, byte[] arr) {
        return getInstance().defineClass(name, arr, 0, arr.length, AsmClassLoader.class.getProtectionDomain());
    }

}

File: module-nms/src/main/kotlin/taboolib/module/nms/AsmClassLoader.kt
package taboolib.module.nms;

public class AsmClassLoader extends ClassLoader {

    private static final class AsmClassLoaderHolder {

        private static final AsmClassLoader INSTANCE = new AsmClassLoader();
    }

    public static AsmClassLoader getInstance() {
        return AsmClassLoaderHolder.INSTANCE;
    }

    private AsmClassLoader() {
        super(AsmClassLoader.class.getClassLoader());
    }

    public static Class<?> createNewClass(String name, byte[] arr) {
        return getInstance().defineClass(name, arr, 0, arr.length, AsmClassLoader.class.getProtectionDomain());
    }
}
